kernel/src/main/java/org/kframework/parser/inner/kernel/Scanner.java

```diff
@@ -50,7 +50,7 @@
  */
 public class Scanner implements AutoCloseable {
 
-    private final Map<TerminalLike, Tuple2<Integer, Integer>> tokens;
+    public final Map<TerminalLike, Tuple2<Integer, Integer>> tokens;
```

Comment:
you need a getter here too that wraps with unmodifiableMap because tokens is a mutable Map.